### PR TITLE
Docker Registry v1 and v2

### PR DESCRIPTION
**Pull request by cmheisel**
Adds Docker registry v2
Retrains Docker registry v1
Updated README

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -20,6 +20,11 @@ docker_registry_dir: /opt/docker-registry
 docker_registry_port: 5000
 docker_registry_vhost_name: "registry registry.yourdomain.local"
 
+docker_registry_2_enable: true
+docker_registry_2_dir: /opt/docker-registry
```
Comment (tehranian) [inline defaults/main.yml]: Is it a problem that the v1 and v2 registries are both enabled by default and are both using the same storage dir?

File: meta/main.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ galaxy_info:
   author: Dan Tehranian
   company: Delphix
   description: Install a collection of caching proxy solutions
-  min_ansible_version: 1.8
+  min_ansible_version: 1.9
```
Comment (tehranian) [inline meta/main.yml]: AFAICT - This still works with Ansible 1.8: https://travis-ci.org/delphix/ansible-package-caching-proxy/builds/90614614

File: defaults/main.yml (hunk 1 above)
Comment (cmheisel) [inline defaults/main.yml]: Good catch, I'll update

File: meta/main.yml (hunk 2 above)
Comment (cmheisel) [inline meta/main.yml]: Sorry I was trying to increment the version of the role, not ansible requirement

**Comment by tehranian** [comment]
@cmheisel Really sorry to do this since you've already re-worked your change, but...

In my comments in #7, I was merely asking what you _thought_ about how to handle the v1/v2 Registry upgrade logistics. Looking at the two diffs and thinking about the longer-term implications, I kind of like the diff from #7 better. If we're going to implement a v2 registry, I think the v1 registry can be removed.

Can we re-open #7, update it with the README changes, then I'll merge it?

Thanks and sorry for any miscommunications on my part.

**Comment by tehranian** [comment]
@cmheisel - Ping? Can we re-open #7, update it with the README changes, then I'll merge it?

**Pull request closed without merging**